@utahdts/utah-design-system/react/components/popups/Modal/Modal.jsx

```diff
@@ -0,0 +1,86 @@
+// @ts-check
+import React, { useCallback, useEffect, useRef } from 'react';
+import { joinClassNames } from '../../../util/joinClassNames';
+import { ICON_BUTTON_APPEARANCE } from '../../../enums/buttonEnums';
+import { IconButton } from '../../buttons/IconButton';
+import { useAriaMessaging } from '../../../contexts/UtahDesignSystemContext/hooks/useAriaMessaging';
+
+/**
+ * @param {object} props
+ * @param {React.ReactNode} [props.children]
+ * @param {string} [props.className]
+ * @param {string} props.id
+ * @param {React.Ref<HTMLDivElement>} [props.innerRef]
+ * @param {import('react').KeyboardEventHandler} [props.onEscape]
+ * @param {import('react').MouseEventHandler} [props.onClose]
+ * @returns {React.JSX.Element}
+ */
+export function Modal({
+  children,
+  className,
+  id,
+  innerRef,
+  onEscape,
+  onClose,
+}) {
+  const ref = /** @type {typeof useRef<HTMLDialogElement>} */ (useRef)(null);
```

Comment:
{typeof useRef<HTMLDialogElement | null>} <-- needs `| null` because it starts as null

Reply:
Another way:
```const ref = useRef(/** @type {HTMLDialogElement | null} */ (null))```